src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1150,198 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
             }
         }
 
         /// <summary>
-        /// Helper method to get the role with which the GraphQL API request was executed.
+        /// Helper method to collect names of all the fields referenced from every entity in a GraphQL mutation.
         /// </summary>
-        /// <param name="context">HotChocolate context for the GraphQL request</param>
-        private static string GetRoleOfGraphQLRequest(IMiddlewareContext context)
+        /// <param name="entityToFields">Dictionary to store all the entities and their corresponding fields referenced in the mutation.</param>
+        /// <param name="schema">Schema for the input field.</param>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="context">Middleware Context.</param>
+        /// <param name="parameters">Value for the input field.</param>
+        /// <example>       1. mutation {
+        ///                 createbook(
+        ///                     item: {
+        ///                         title: "book #1",
+        ///                         reviews: [{ content: "Good book." }, { content: "Great book." }],
+        ///                         publisher: { name: "Macmillan publishers" },
+        ///                         authors: [{ birthdate: "1997-09-03", name: "Red house authors", author_name: "Dan Brown" }]
+        ///                     })
+        ///                 {
+        ///                     id
+        ///                 }
+        ///                 2. mutation {
+        ///                 createbooks(
+        ///                     items: [{
+        ///                         title: "book #1",
+        ///                         reviews: [{ content: "Good book." }, { content: "Great book." }],
+        ///                         publisher: { name: "Macmillan publishers" },
+        ///                         authors: [{ birthdate: "1997-09-03", name: "Red house authors", author_name: "Dan Brown" }]
+        ///                     },
+        ///                     {
+        ///                         title: "book #2",
+        ///                         reviews: [{ content: "Awesome book." }, { content: "Average book." }],
+        ///                         publisher: { name: "Pearson Education" },
+        ///                         authors: [{ birthdate: "1990-11-04", name: "Penguin Random House", author_name: "William Shakespeare" }]
+        ///                     }])
+        ///                 {
+        ///                     items{
+        ///                         id
+        ///                         title
+        ///                     }
+        ///                 }</example>
+        private void PopulateMutationEntityAndFieldsToAuthorize(
+            Dictionary<string, HashSet<string>> entityToFields,
+            IInputField schema,
+            string entityName,
+            IMiddlewareContext context,
+            object? parameters)
```

Comment:
I think, it should be fine to keep it non-nullable. I'd need to add another conditional for null values which I think would be more clear rather than making this param nullable.